http-client/src/main/java/io/airlift/http/client/jetty/JettyHttpClient.java

```diff
@@ -137,6 +159,9 @@ private JettyHttpClient(HttpClientConfig config, Optional<JettyIoPool> jettyIoPo
         maxContentLength = config.getMaxContentLength().toBytes();
         requestTimeoutMillis = config.getRequestTimeout().toMillis();
         idleTimeoutMillis = config.getIdleTimeout().toMillis();
+        this.securityConfig = securityConfig;
+        authenticationProvider = securityConfig != null && securityConfig.enabled() ?
+                new AuthenticationProvider(securityConfig) : null;
```

Comment:
We shouldn't allow securityConfig to be null. This class should always expect a valid instance.